### ISSUE TITLE
[CI] MultiFeatureMigrationIT testMigrateInternalManagedSystemIndex failing

### ISSUE THREAD

**Issue by bpintea** [elastic/elasticsearch#79894]
**Build scan:**
https://gradle-enterprise.elastic.co/s/56rmzpaa7gopo/tests/:modules:reindex:internalClusterTest/org.elasticsearch.migration.MultiFeatureMigrationIT/testMigrateInternalManagedSystemIndex

**Reproduction line:**
`./gradlew ':modules:reindex:internalClusterTest' --tests "org.elasticsearch.migration.MultiFeatureMigrationIT.testMigrateInternalManagedSystemIndex" -Dtests.seed=7F41A39C4FCAC164 -Dtests.locale=sl-SI -Dtests.timezone=Asia/Yangon -Druntime.java=8`

**Applicable branches:**
7.16

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.migration.MultiFeatureMigrationIT&tests.test=testMigrateInternalManagedSystemIndex

**Failure excerpt:**
```
java.lang.NullPointerException: (No message provided)

  at __randomizedtesting.SeedInfo.seed([7F41A39C4FCAC164:20E3CC05093EB676]:0)
  at org.elasticsearch.migration.FeatureMigrationIT.assertIndexHasCorrectProperties(FeatureMigrationIT.java:220)
  at org.elasticsearch.migration.FeatureMigrationIT.testMigrateInternalManagedSystemIndex(FeatureMigrationIT.java:173)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

```

**Comment by williamrandolph** [comment]
These are some fairly new system index tests. It looks like we have some race conditions. We fixed one known race condition (https://github.com/elastic/elasticsearch/pull/80049) today, so there's a chance we caught this failure... but if this fails again, I will be happy to mute.

**Comment by ywangd** [comment]
One more on 7.16 https://gradle-enterprise.elastic.co/s/ftowooeim7ozi/tests/:modules:reindex:internalClusterTest/org.elasticsearch.migration.MultiFeatureMigrationIT/testMigrateInternalManagedSystemIndex?top-execution=1

**Comment by javanna** [comment]
One more failure: https://gradle-enterprise.elastic.co/s/snx4uzytzy3kk/tests/:modules:reindex:internalClusterTest/org.elasticsearch.migration.MultiFeatureMigrationIT/testMigrateInternalManagedSystemIndex?top-execution=1

**Comment by martijnvg** [comment]
Another instance of this test failure: https://gradle-enterprise.elastic.co/s/lshlnnmjyokcw

In the same build failure also other tests fail, but I'm not sure whether these test failures are related to this on or require a separate issue to be opened. Other failing test method are: testMultipleFeatureMigration() and testStartMigrationAndImmediatelyCheckStatus().

**Comment by ywangd** [comment]
Again on 7.16 https://gradle-enterprise.elastic.co/s/q6gggnydnbn4s/tests/:modules:reindex:internalClusterTest/org.elasticsearch.migration.FeatureMigrationIT/testMigrateInternalManagedSystemIndex?expanded-stacktrace=WyIwIl0

**Comment by martijnvg** [comment]
A different test failure for the same test: https://gradle-enterprise.elastic.co/s/vj2jmywbmshag#tests
(also other tests in the same suite failed).

```
java.lang.AssertionError: |  
-- | --
  | Expected: <NO_MIGRATION_NEEDED> |  
  | but: was <IN_PROGRESS>


		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)	
		at org.junit.Assert.assertThat(Assert.java:956)	
		at org.junit.Assert.assertThat(Assert.java:923)	
		at org.elasticsearch.migration.FeatureMigrationIT.lambda$testMigrateInternalManagedSystemIndex$5(FeatureMigrationIT.java:198)	
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1140)	
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1113)	
		at org.elasticsearch.migration.FeatureMigrationIT.testMigrateInternalManagedSystemIndex(FeatureMigrationIT.java:194)
```

The this failure somehow related to the NPE mentioned in this issue? If not, should this failure be tracked in a different  test failure issue?

**Comment by williamrandolph** [comment]
I think this was duplicated by https://github.com/elastic/elasticsearch/issues/82787 and, I hope, fixed by #84646.
